# PROBLEM STATEMENT:DETECTION OF SPAM EMAILS

# This is a simple Machine Learning project for beginners that distinguishes whether a mail is spam or not.

In [1]:
import pandas as pd

In [2]:
spam=pd.read_csv('spam_dataset.csv')

In [3]:
spam.head()

id label                                               text  label_num
0   605   ham  Subject: enron methanol ; meter # : 988291\r\n...          0
1  2349   ham  Subject: hpl nom for january 9 , 2001\r\n( see...          0
2  3624   ham  Subject: neon retreat\r\nho ho ho , we ' re ar...          0
3  4685  spam  Subject: photoshop , windows , office . cheap ...          1
4  2030   ham  Subject: re : indian springs\r\nthis deal is t...          0

In [4]:
spam=spam.drop(['id'],axis=1)
spam.head()

label                                               text  label_num
0   ham  Subject: enron methanol ; meter # : 988291\r\n...          0
1   ham  Subject: hpl nom for january 9 , 2001\r\n( see...          0
2   ham  Subject: neon retreat\r\nho ho ho , we ' re ar...          0
3  spam  Subject: photoshop , windows , office . cheap ...          1
4   ham  Subject: re : indian springs\r\nthis deal is t...          0

In [5]:
spam.shape

(5171, 3)

In [6]:
labels=spam.label
labels.head()

0     ham
1     ham
2     ham
3    spam
4     ham
Name: label, dtype: object

In [7]:
from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.linear_model import PassiveAggressiveClassifier
from sklearn.metrics import accuracy_score, confusion_matrix

In [20]:
x_train,x_test,y_train,y_test=train_test_split(spam['text'],labels,test_size=0.3,random_state=42)

In [21]:
tfidf=TfidfVectorizer(stop_words='english',max_df=0.7)
x_train=tfidf.fit_transform(x_train)
x_test=tfidf.transform(x_test)

In [22]:
pac=PassiveAggressiveClassifier(max_iter=180,random_state=42)
pac.fit(x_train,y_train)
pred=pac.predict(x_test)

In [23]:
print(accuracy_score(pred,y_test))

0.9942010309278351


In [25]:
print('Percentage Accuracy:')
print(round(accuracy_score(pred,y_test)*100,2))

Percentage Accuracy:
99.42


# An accuracy of 99.4% has been acieved,which I believe is a good level of accuracy.